solr/core/src/java/org/apache/solr/packagemanager/SolrPackage.java

```diff
@@ -0,0 +1,107 @@
+package org.apache.solr.packagemanager;
+
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Describes a package (along with all released versions) as it appears in a repository.
+ */
+public class SolrPackage implements Serializable, Comparable<SolrPackage> {
+
+  public String id;
+  public String description;
+  public List<SolrPackageRelease> versions;
+
+  private String repositoryId;
+
+  public static class SolrPackageRelease {
+
+    public String version;
+    public Date date;
+    public String url;
+
+    public String sha512sum;
+    public String sig;
+
+    public Metadata metadata;
+    @Override
+    public String toString() {
```

Comment:
implement MapWriter, and toString can be using  `jsonStr()`

or even better , implement `ReflectMapWriter` and annotate your fields using `@JsonProperty`

Reply:
Done, thanks!